sympy/solvers/tests/test_ode.py

```diff
@@ -2285,7 +2285,9 @@ def test_nth_linear_constant_coeff_variation_of_parameters_simplify_False():
     sol_simp = dsolve(eq, f(x), hint=our_hint, simplify=True)
     sol_nsimp = dsolve(eq, f(x), hint=our_hint, simplify=False)
     assert sol_simp != sol_nsimp
-    assert checkodesol(eq, sol_simp, order=5, solve_for_func=False)[0]
+    # eq.subs(*sol_simp.args) doesn't simplify to zero without help
+    zero = checkodesol(eq, sol_simp, order=5, solve_for_func=False)[1]
+    assert zero.rewrite(exp).simplify() == 0
```

Comment:
```suggestion
    # if this fails because zero.is_zero, replace this and preceding with
    # assert checkodesol(eq, sol_simp, order=5, solve_for_func=False)[0]
    assert not zero.is_zero and zero.rewrite(exp).simplify() == 0
```